pints/_transformation.py

```diff
@@ -0,0 +1,1120 @@
+        along with the partial derivatives of the result with respect to the
+        parameters.
+
+        The returned data is a tuple ``(S, S')`` where ``S`` is a
+        ``n_parameters`` by ``n_parameters`` matrix and ``S'`` is a sequence of
+        ``n_parameters`` matrices.
+
+        *This is an optional method.* It is needed when the transformation is
+        used along with a non-element-wise transformation in
+        :class:`ComposedTransformation`.
+        """
+        raise NotImplementedError
+
+    def log_jacobian_det(self, q):
+        """
+        Returns the logarithm of the absolute value of the determinant of the
+        Jacobian matrix of the transformation :meth:`Transformation.jacobian`
+        calculated at the parameter vector ``q`` in the search space.
+
+        *This is an optional method.* It is needed when transformation is
+        performed on :class:`LogPDF` and/or that requires ``evaluateS1()``;
+        e.g. not necessary if it's used for :class:`ErrorMeasure` without
+        :meth:`ErrorMeasure.evaluateS1()`.
+        """
+        return np.log(np.abs(np.linalg.det(self.jacobian(q))))
+
+    def log_jacobian_det_S1(self, q):
+        """
+        Computes the logarithm of the absolute value of the determinant of the
+        Jacobian, and returns the result plus the partial derivatives of the
+        result with respect to the parameters.
+
+        The returned data is a tuple ``(S, S')`` where ``S`` is a scalar value
+        and ``S'`` is a sequence of length ``n_parameters``.
+
+        Note that the derivative returned is of the log of the determinant of
+        the Jacobian, so ``S' = d/dq log(|det(J(q))|)``, evaluated at input.
+
+        *This is an optional method.* It is needed when transformation is
+        performed on :class:`LogPDF` and that requires ``evaluateS1()``.
+        """
+        # Directly calculate the derivative using jacobian_S1()
+        #
+        # d/dq log(|det(J(q))|) = 1/|det(J(q))| * sign(det(J(q)))
+        #                         * d/dq det(J(q))
+        #
+        # The second term is given by Eq. 46 in the Matrix Cookbook (2012)
+        # http://www.math.uwaterloo.ca/~hwolkowi/matrixcookbook.pdf
+        #
+        # d/dq det(J(q)) = det(J) Tr(J^{-1} d/dq J)
+        #
+        # Therefore
+        #
+        # d/dq log(|det(J(q))|) = Tr(J^{-1} d/dq J)
+        #
+        q = pints.vector(q)
+        jac, jac_S1 = self.jacobian_S1(q)
+        out_S1 = np.zeros(q.shape)
+        for i, jac_S1_i in enumerate(jac_S1):
+            out_S1[i] = np.trace(np.matmul(np.linalg.pinv(jac), jac_S1_i))
+        return self.log_jacobian_det(q), out_S1
+
+    def multiple_to_model(self, qs):
+        """
+        Transforms a series of parameter vectors ``qs`` from the search space
+        to the model space. ``qs`` must be provided in the shape
+        (``n_vectors``, ``n_parameters``).
+        """
+        qs = np.asarray(qs)
```

Comment:
If we're going to use this a lot (and it doesn't disappear, see comments above), then maybe do a quick benchmark to see if it's faster as a list comprehension (with an array wrapped around that)

Reply:
Removed this method.